multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -832,7 +801,6 @@ def fit(
             time_limit = timedelta(seconds=time_limit)
 
         # set attributes for saving and prediction
-        self._problem_type = problem_type  # In case problem type isn't provided in __init__().
```

Comment:
Wondering why do we need to move this line to above?

Reply:
Set self._problem_type right after we got the inferred problem_type. This will avoid potential misuse of self._problem_type.